src/briefcase/integrations/adb.py

```diff
@@ -0,0 +1,27 @@
+import subprocess
+import os
+
+from briefcase.exceptions import BriefcaseCommandError
+
+
+def get_devices(sdk_path, sub=subprocess):
+    # Add our `adb` to start of the path.
+    env = dict(os.environ)
+    env['PATH'] = str(sdk_path / 'platform-tools') + ':' + env.get('PATH', '')
+    try:
+        output = sub.check_output(['adb', 'devices'], env=env)
```

Comment:
Do we want to invoke `adb devices -l` and parse more details? From my testing, `adb devices` tells you:
```
List of devices attached
04ea55ee8292009a	device
KAPHN4G9AA1513	device
emulator-5554	device
```
whereas `adb devices -l` gives you:
```
04ea55ee8292009a       device usb:340869888X product:hammerhead model:Nexus_5 device:hammerhead transport_id:4
KAPHN4G9AA1513         device usb:340870144X product:Kogan_Agora_9 model:Kogan_Agora_9 device:Kogan_Agora_9 transport_id:3
emulator-5554          device product:sdk_phone_x86 model:Android_SDK_built_for_x86 device:generic_x86 transport_id:2
```
That would also allow the UI to present a more helpful list of choices when there are multiple devices available.

Reply:
Another useful data point: if a device is *not* authorized for debugging mode, but *is* plugged in, you get:
```
List of devices attached
04ea55ee8292009a       unauthorized usb:340869888X transport_id:4
KAPHN4G9AA1513         device usb:340870144X product:Kogan_Agora_9 model:Kogan_Agora_9 device:Kogan_Agora_9 transport_id:3
emulator-5554          device product:sdk_phone_x86 model:Android_SDK_built_for_x86 device:generic_x86 transport_id:2
```

Reply:
Oh, huh. This is a good idea. +1